src/DI/ContainerBuilder.php

```diff
@@ -55,16 +55,17 @@ public function __construct()
 
 	/**
 	 * Adds new service definition.
-	 * @return Definitions\ServiceDefinition
+	 * @return Definition|Definitions\ServiceDefinition
```

Comment:
Definition is subclass of ServiceDefinition

Reply:
I tried to change it and got reported `Call to an undefined method Nette\DI\Definitions\Definition::setFactory().`. Which is true, but this would break every single repo using phpstan. Better not to fix, same as for change in FactoryDefinition

Reply:
Dark side of a BC :/ 
`addDefinition(Definition $definition, ?string $name = null): void` would work perfectly.

Reply:
I agree that this should not be removed until a solution for 3rd code is available.

Reply:
Removed.